sklearn/compose/_column_transformer.py

```diff
@@ -686,8 +687,22 @@ def _get_transformer_list(estimators):
     Construct (name, trans, column) tuples from list
 
     """
-    transformers = [trans[1] for trans in estimators]
-    columns = [trans[0] for trans in estimators]
+    # check the first tuple in estimators, swap all if it
+    # follows the (columns, transformer) order.
+    # remove in v0.22
+    trans_ix, cols_ix = 0, 1
+    if (hasattr(estimators[0][1], 'fit') or
+            (estimators[0][1] in ('drop', 'passthrough')
+             and estimators[0][0] not in ('drop', 'passthrough'))):
```

Comment:
Now you're not catching the (obscure) case ``make_column_transformer(("passthrough", "passthrough"), (columns, transrformer))``

Reply:
right, I guess this is now clear.